libs/utils/src/split-props.ts

```diff
@@ -0,0 +1,23 @@
+import { String, pipe } from 'effect'
+import type { SplitProps } from './typescript'
+
+export const splitProps = <Props, Mixins extends string[]>(props: Props, ...mixins: Mixins) => {
```

Comment:
**thinking:** I wonder about performance impact of this change. Given the re-rendering nature of React it might be a problem. Maybe it could be abstracted into useMemo and defined as hook?

Reply:
extracting it into useMemo doesn't solve the performance problem, actually spent over 2hrs trying to decipher that without success :D

cuz, when you memoize splitProps, you definetly need to add the `props` parameter to the list of dependency so that it reflects the up-to-date state in the component where it's being used, and that's where the problem is.
when you add that to the list of dependency, it's just as though you had the same function without the hook and useMemo cuz it gets re-calculated every single time the props change.

I think it's one of the trade offs to the pattern that we need to think about if we decide to move forward with it.

We just need to be a bit careful when using it,  for example not adding events that causes a component to re-render rapidly, 

Otherwise, We're good with the current implementation

BTW, I've seen a CRM platform use it, even with most of their components being controlled yet it's not something they're scared about.